src/main/java/org/mondemand/Client.java

```diff
@@ -1267,7 +1281,6 @@ private synchronized void dispatchStatsSamples() {
 
       // snapshot samples map for dispatch
       SamplesMessage[] samplesMsgs = this.samples.values().toArray(new SamplesMessage[0]);
-      this.samples = new ConcurrentHashMap<String, SamplesMessage>();
```

Comment:
I removed this line because it seemed unnecessary. dispatchStatsSamples is only called in flush() where this.samples is cleared anyways.